contrib/intel/jenkins/tests.py

```diff
@@ -0,0 +1,309 @@
+import sys
+import os
+
+print(os.environ['CI_SITE_CONFIG'])
+sys.path.append(os.environ['CI_SITE_CONFIG']) # for adding path for ci_site_config
+
+import subprocess
+import re
+import ci_site_config
+import common
+import shlex
+
+class Test:
+    def __init__ (self, branchname, buildno, testname, core_prov, fabric,
+                  hosts, ofi_build_mode, util_prov=None):
+        self.branchname = branchname
+        self.buildno = buildno
+        self.testname = testname
+        self.core_prov = core_prov
+        self.util_prov = "ofi_{}".format(util_prov) if util_prov != None else "" 
+        self.fabric = fabric
+        self.hosts = hosts
+        self.ofi_build_mode = ofi_build_mode
+        if (len(hosts) == 2):
+            self.server = hosts[0]
+            self.client = hosts[1]
+       
+        self.nw_interface = ci_site_config.interface_map[self.fabric]
+        self.libfab_installpath = "{}/{}/{}/{}".format(ci_site_config.install_dir,
+                                  self.branchname, self.buildno, self.ofi_build_mode)
+ 
+        self.env = [("FI_VERBS_MR_CACHE_ENABLE", "1"),\
+                    ("FI_VERBS_INLINE_SIZE", "256")] \
+                    if self.core_prov == "verbs" else []
+    
+     
+
+class Fabtest(Test):
+    
+    def __init__(self, branchname, buildno, testname, core_prov, fabric,
+                 hosts, ofi_build_mode, util_prov=None):
+        
+        super().__init__(branchname, buildno, testname, core_prov, fabric,
+                         hosts, ofi_build_mode, util_prov)
+        self.fabtestpath = "{}/bin".format(self.libfab_installpath) 
+    
+    def get_exclude_file(self):
+        path = self.libfab_installpath
+        efile_path = "{}/share/fabtests/test_configs".format(path)
+
+        prov = self.util_prov if self.util_prov else self.core_prov
+        efile_old = "{path}/{prov}/{prov}.exclude".format(path=efile_path, 
+                      prov=prov)
+        
+        if self.util_prov:
+            efile = "{path}/{util_prov}/{core_prov}/exclude".format(path=efile_path,
+                      util_prov=self.util_prov, core_prov=self.core_prov)
+        else:
+            efile = "{path}/{prov}/exclude".format(path=efile_path,
+                      prov=self.core_prov)
+           
+        if os.path.isfile(efile):
+            return efile
+        elif os.path.isfile(efile_old):
+            return efile_old
```

Comment:
The efile_old is not needed in the latest versions. I think it's needed for v1.6 or older. This change will be needed there if you ever backport Jenkinsfile to v1.6

Reply:
wouldn't it be better to retain it than remove it. it may be harder to trace back the changes required later in case back porting is required? whereas at the moment it will work for both the cases.

Reply:
Yes, it will work for both cases if you retain it. It's just unnecessary code to maintain.